### DISCUSSION TITLE
wg-easy on Synology working but no access to files on Synology

### DISCUSSION THREAD

**Discussion by Vinz3ntR** [wg-easy/wg-easy#598] [Q&A] [answered]
I installed wg-easy on my Synology NAS. It works, I can connect to it and use my home internet connection. I can access any of my devices; PC, IP camera's and it's really stable and fast.

But, I need to access my files. That basically was the whole point. I can's acces my files on the NAS and also not the main page. Basically anything on the NAS itself where Docker is running is not working.

Is there a way to fix that?

**Comment by DonPablo1010** [comment, marked as answer]
You can assign a separate IP address to wg-easy using macvlan and then add a macvlan bridge to it from your NAS with a virtual IP that is different from its original IP. This setup allows both IPs to communicate properly in both directions, even when running within a Docker container. I’ve successfully done this with AdGuardHome, enabling me to use both Synology’s DNS server and AdGuard in a sequential manner.

In this example, your NAS has the IP address 192.168.1.10.

1. Build a separate bridge network in Docker that differs from the default bridge and doesn’t use the host network. For instance, you can create a new network named “synobridge” with the following settings: driver bridge, subnet 172.18.0.0/16, gateway 172.18.0.1.

2. Log in to your NAS and execute the command to create a new macvlan, giving wg-easy its own IP address. First, identify your interface name using the `ip route` command. In our example, the interface is eth0, and the desired IP for wg-easy is 192.168.1.55. Here’s the command to run via SSH:

`sudo docker network create -d macvlan --subnet=192.168.1.0/24 --ip-range=192.168.1.55/32 --gateway=192.168.1.1 -o parent=eth0 wg-easy_fixip`

3. Create the wg-easy Docker container and add the newly created “synobridge” network as well as the “wg-easy_fixip” network. Now, wg-easy has its own IP address, 192.168.1.55, and is reachable under that IP.

4. Create a macvlan bridge with an IP address one above your NAS’s IP to allow your host to communicate with wg-easy’s IP. You can achieve this using the Synology Task Scheduler. Add the following script as a task to run with the root user on any reboot:

```
#!/bin/sh

if ip link | grep "wg-easy-connect@eth0" > /dev/null; then
    echo "Device wg-easy-connect exists"
else
    echo "Create Device wg-easy-connect"
    ip link add wg-easy-connect link eth0 type macvlan mode bridge
    ip addr add 192.168.1.11/32 dev wg-easy-connect
    ip link set wg-easy-connect up
    ip route add 192.168.1.55/32 dev wg-easy-connect
fi
```
